src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -1194,6 +1195,77 @@ export class TerminalService extends Disposable implements ITerminalService {
 	setEditingTerminal(instance: ITerminalInstance | undefined) {
 		this._editingTerminal = instance;
 	}
+
+	createInstanceEventMultiplexer<T>(getEvent: (instance: ITerminalInstance) => Event<T>): { dispose(): void; event: Event<T> } {
+		const store = new DisposableStore();
+		const multiplexer = store.add(new EventMultiplexer<T>());
+		const instanceListeners = store.add(new DisposableMap<ITerminalInstance, IDisposable>());
+
+		function addInstance(instance: ITerminalInstance) {
+			const listener = multiplexer.add(getEvent(instance));
+			instanceListeners.set(instance, listener);
+		}
+
```

Comment:
You could name the variables accordingly to delete the comments like `for (const existingInstance of this.instances)`